### PR TITLE
add dilated convolution.

### PR DESCRIPTION
**Pull request by YoshikawaMasashi**
I implemented Dilated CNN in magenta and made it available for generating melody. Dilated CNN is a model used for generation of time series such as Wavenet and shows good performance in translation and sound generation.

### CODE REVIEW

**Comment by YoshikawaMasashi** [comment]
I tried to agree to the CLA (Google Individual Contributor License Agreement), but at that time I got an error message "You must be an owner of the contributors group in order to submit this CLA".
What is the contributors group?
And how can I submit this CLA?

**Comment by YoshikawaMasashi** [comment]
It solved.
I signed it!

**Comment by YoshikawaMasashi** [comment]
Sorry.
I signed it!

**Comment by cghawthorne** [comment]
Hi @YoshikawaMasashi, this looks really interesting! Do you have any example outputs of this model?

**Comment by YoshikawaMasashi** [comment]
Hi!
You can hear the melody generated by this option(primer_melody="[60,60,-1,60,59,60,60,60,-2,60,64,-2,62,-2,60,-2]") in soundcloud.
https://soundcloud.com/ig4osq8tqokz/dilatedcnnmagentademo

File: magenta/models/shared/events_rnn_graph.py [hunk 1]
```diff
@@ -52,6 +52,175 @@ def make_rnn_cell(rnn_layer_sizes,
...
+        after = tf.nn.conv2d(
+            gate*filtr, after_weights,strides=[1,1,1,1],padding='SAME')
+      else:
+        after = tf.nn.conv2d(
+            filtr, after_weights,strides=[1,1,1,1],padding='SAME')
+        after = tf.nn.bias_add(after, after_biases)
+
+      if use_step:
+        step.append(after)
+        h = after + _h
```
Comment (erkyz) [inline magenta/models/shared/events_rnn_graph.py]: Shouldn't this line be outside of the conditional?

Comment (YoshikawaMasashi) [inline magenta/models/shared/events_rnn_graph.py]: Thank you for your review very much.
I made a mistake.
I modified the code.

**Pull request closed without merging**